### PR TITLE
feat: add lending page components caching

### PR DESCRIPTION
**Pull request by edwinf-barreram**
# Description
- Added cache function to Lending page components

# Ticket
[MARS-131](https://kiva.atlassian.net/browse/MARS-131)

### CODE REVIEW

File: src/pages/Lend/LendHeader.vue [hunk 1]
```diff
@@ -49,6 +50,19 @@ export default {
...
+			browseUrl,
+			filterUrl,
+			sidePinnedFilterPadding,
+			hardLeftAlign,
+			sideArrowsPadding
+		} = props;
+		const getValue = value => {
+			return value ? `-${value}` : '';
+		};
+		getCacheKey(`${filterUrl}${getValue(browseUrl)}${getValue(sidePinnedFilterPadding)}${getValue(hardLeftAlign)}${getValue(sideArrowsPadding)}`); // eslint-disable-line max-len
```
Comment (mcstover) [inline src/pages/Lend/LendHeader.vue:64]: It seems like the filterUrl + browserUrl shouldn't be used on this component. Rather they would go on the ViewToggle where they would actually differ? Or, does it need to be the parent (this component) that manages that...

I just wonder if either `filterUrl` or `browserUrl` will contain query params and result in a lot of instances of this being cached...

File: package-lock.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
 	"name": "kvue",
-	"version": "2.156.2",
+	"version": "2.157.0",
```
Comment (dyersituations) [inline package-lock.json:3]: Which `npm` version are you running? This change can be reverted.

File: src/components/Kv/KvLoadingOverlay.vue [hunk 3]
```diff
@@ -9,9 +9,11 @@
 <script>
 import _throttle from 'lodash/throttle';
 import KvLoadingSpinner from '@/components/Kv/KvLoadingSpinner';
+import getCacheKey from '@/util/getCacheKey';
 
 export default {
 	name: 'KvLoadingOverlay',
+	serverCacheKey: () => getCacheKey('KvLoadingOverlay'),
```
Comment (dyersituations) [inline src/components/Kv/KvLoadingOverlay.vue:16]: Consider using constants to share strings in these components where they are exact matches, such as `KvLoadingOverlay`, if the name/key match is important here. If the caching is only done based on the `serverCacheKey` and doesn't consider the `name` prop, ignore this comment :).

File: package-lock.json (hunk 2 above)
Comment (edwinf-barreram) [reply package-lock.json:3]: I am using version 8.5.0

Comment (mcstover) [reply package-lock.json:3]: I think this is ok... it will get updated in the background again later...

File: src/pages/Lend/LendHeader.vue (hunk 1 above)
Comment (edwinf-barreram) [reply src/pages/Lend/LendHeader.vue:64]: @mcstover this is a result is that is currently being generated there **_/lend/filter-/lend-by-category-true_**

File: package-lock.json (hunk 2 above)
Comment (dyersituations) [reply package-lock.json:3]: Ah this version will be auto-updated to match the package.json version.

File: src/pages/Lend/LendHeader.vue (hunk 1 above)
Comment (mcstover) [reply src/pages/Lend/LendHeader.vue:64]: Ok suppose that's ok. The "/" characters are a little odd but no query params in site!

**Review by mcstover — APPROVED** [review]

File: src/components/Kv/KvLoadingOverlay.vue (hunk 3 above)
Comment (edwinf-barreram) [reply src/components/Kv/KvLoadingOverlay.vue:16]: Name is being set here because this component does not contains props.

Comment (emuvente) [reply src/components/Kv/KvLoadingOverlay.vue:16]: @dyersituations The name/key match isn't important here. They are both used together to create the final cache key: `${component.name}::${component.serverCacheKey(props)}`

**Review by emuvente — COMMENTED** [review]
Looking good! There are a just a couple changes needed

**Review by mcstover — APPROVED** [review]

**Pull request merged**